### PR TITLE
Add go 1.23 in docker image and remove 1.16-1.18 support.

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -140,6 +140,7 @@ base.image.go1.19 := golang:1.19
 base.image.go1.20 := golang:1.20
 base.image.go1.21 := golang:1.21
 base.image.go1.22 := golang:1.22
+base.image.go1.23 := golang:1.23
```
Comment (wu-sheng) [inline Makefile:143]: I am good to add 1.23 added. 

Meanwhile, according to go maintenance policy `Each major Go release is supported until there are two newer major releases.`. I think we at least only need to release N-4?

Comment (wu-sheng) [reply Makefile:143]: > go1.19.13 (released 2023-09-06) includes fixes to the go command, and the crypto/tls and net/http packages. See the [Go 1.19.13 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.19.13+label%3ACherryPickApproved) on our issue tracker for details.

As N-4(1.19) was not released for nearly a year.

Comment (wu-sheng) [reply Makefile:143]: Mostly, 1.18 should be the lowest version we plan to support.

Comment (wu-sheng) [reply Makefile:143]: I refer to https://go.dev/doc/devel/release

Comment (mrproliu) [reply Makefile:143]: Ok. Then we can remove between the 16-19 versions?

Comment (wu-sheng) [reply Makefile:143]: According to release note, we could at least remove 16, 17, 18, 19(this could consider to be kept).

Comment (mrproliu) [reply Makefile:143]: Sure, I will keep the 19 and higher versions.

Comment (mrproliu) [reply Makefile:140]: I will temporarily make this PR a draft to wait for all testing success because all older tests are based on go `1.18`.

**Comment by wu-sheng** [comment]
Do we have a doc to indicate lowest support version? We should update it.

**Comment by mrproliu** [comment]
> Do we have a doc to indicate lowest support version? We should update it.

Yes, I have updated.

**Comment by wu-sheng** [comment]
About native http, the doc says supporting to 1.20.
Do we support that to 1.23? As it is a native lib.

**Comment by mrproliu** [comment]
> About native http, the doc says supporting to 1.20.
> Do we support that to 1.23? As it is a native lib.

Done, all native frameworks have added tests for 1.23.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**